src/api/middlewares/Authentication.ts

```diff
@@ -69,6 +70,16 @@ declare global {
 	}
 }
 
+let jwtPublicKey: KeyObject;
+
+// Initialize the jwt secret as a key object so it does not need to be regenerated for each request.
+export function initAuthentication(api: Router) {
```

Comment:
checkToken is called in a few other places like gateway or for email verification. those can't access this

Reply:
how much does this improve perf by, anyway?

Reply:
A few milliseconds per request, which add up if there are many requests

Reply:
The checkToken function still works with strings, but is faster when passing a KeyObject